### PR TITLE
changed camel_case.rs to a regex based function

### PR DESCRIPTION
**Pull request by Csatinapppple**
It retains the String input and output of the function, might change it to &str if needed

### CODE REVIEW

File: src/camel_case.rs [hunk 1]
```diff
@@ -4,36 +4,34 @@
...
-                    index = index + 1;
-                    continue;
-                }
-                str_array[index] = str_array[index].to_ascii_uppercase();
-                break;
-            }
-            continue;
-        }
-        combine_result.push_str(&char_to_string)
+pub fn new(mut string: String) -> String{
```
Comment (charleslukes) [inline src/camel_case.rs:11]: make this a pure function to avoid mutating user original string value.

File: src/camel_case.rs [hunk 2]
```diff
@@ -4,36 +4,34 @@
...
+    if ret.chars().next().map(|c|!c.is_ascii()).unwrap_or(true){
+        return ret.to_string();
     }
+    let tail = &ret[1..];
 
-    return combine_result;
+    format!("{}{tail}",ret.chars().next().unwrap().to_lowercase())
 }
 
+// TODO: Rewrite just using regex
```
Comment (charleslukes) [inline src/camel_case.rs:34]: comment no longer needed.

File: src/camel_case.rs [hunk 3]
```diff
@@ -4,36 +4,34 @@
...
+    //regex to substitute
+    let re = regex::Regex::new(r"(?x)
+        (?P<cut>[^a-zA-Z\d]+) # every char not in camelCase
+        (?:(?P<replace>
+            \d+[a-zA-Z]  # example: 24h
+            |
+            \d+    # example: 500
+            |
+            [a-zA-Z]  # example: format
+        )|$)").unwrap();
```
Comment (charleslukes) [inline src/camel_case.rs:21]: awesome regex here! what formatter do you use? run `cargo fmt` to properly format

**Comment by charleslukes** [comment]
This looks very good 👍. Once the comments are resolved I'll merge.

File: src/camel_case.rs (hunk 1 above)
Comment (Csatinapppple) [reply src/camel_case.rs]: Created a prep function to copy and run .to_lowercase()

File: src/camel_case.rs (hunk 2 above)
Comment (Csatinapppple) [reply src/camel_case.rs]: removed

File: src/camel_case.rs (hunk 3 above)
Comment (Csatinapppple) [reply src/camel_case.rs]: ran through cargo fmt

**Review by charleslukes — APPROVED** [review]

**Pull request merged**